plugins/modules/foreman_provisioning_template.py

```diff
@@ -360,59 +338,48 @@ def main():
             module.fail_json(
                 msg='No name specified and no filename to infer it.')
 
-    name = entity_dict['name']
-
-    affects_multiple = name == '*'
+    affects_multiple = entity_dict['name'] == '*'
     # sanitize user input, filter unuseful configuration combinations with 'name: *'
     if affects_multiple:
-        if state == 'present_with_defaults':
+        if module.state == 'present_with_defaults':
             module.fail_json(msg="'state: present_with_defaults' and 'name: *' cannot be used together")
-        if state == 'absent':
+        if module.desired_absent:
             if len(entity_dict.keys()) != 1:
                 module.fail_json(msg="When deleting all templates, there is no need to specify further parameters.")
 
     module.connect()
 
-    try:
-        if affects_multiple:
-            entities = find_entities(ProvisioningTemplate)
-        else:
-            entities = find_entities(ProvisioningTemplate, name=entity_dict['name'])
-    except Exception as e:
-        module.fail_json(msg='Failed to search for entities: %s ' % e)
+    if affects_multiple:
```

Comment:
Thinking aloud: I've seen this block multiple times now, time for a helper func?

Reply:
Apropos growing out of scope. I think this can be another pr (without rerecording, if done right).

Reply:
Sure, we even have an old issue for that (https://github.com/theforeman/foreman-ansible-modules/issues/244) from when y'all added that multiple logic.